jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/mapper/PostMapperDecorator.java

```diff
@@ -30,6 +42,104 @@ public Post postRequestToEntity(PostRequest postRequest, String userName) {
+                        .filter(postComment -> !updatePostCommentsRequest.contains(postComment))
+                        .toList();
+                postCommentsToRemove.forEach(post::removeComment);
+
+                // Update the existing database rows which can be found
+                // in the incoming collection (updateCustomerRequest.getOrders())
+                List<PostComment> newPostComments = updatePostCommentsRequest.stream()
+                        .filter(postComment -> !post.getComments().contains(postComment))
+                        .toList();
+
+                updatePostCommentsRequest.stream()
+                        .filter(postComment -> !newPostComments.contains(postComment))
+                        .forEach((postComment) -> {
+                            postComment.setPost(post);
+                            PostComment mergedPostComment = this.postCommentRepository.save(postComment);
+                            post.getComments().set(post.getComments().indexOf(mergedPostComment), mergedPostComment);
+                        });
+
+                // Add the rows found in the incoming collection,
+                // which cannot be found in the current database snapshot
+                newPostComments.forEach(post::addComment);
+            } else {
+                List<PostComment> comments = new ArrayList<>(post.getComments());
+                for (PostComment comment : comments) {
+                    post.removeComment(comment);
+                }
+            }
+        } else {
+            addPostCommentsToPost(postRequest.comments(), post);
+        }
+        if (!CollectionUtils.isEmpty(post.getTags())) {
+            if (!CollectionUtils.isEmpty(postRequest.tags())) {
+
+                // convertPostCommentsDTO to PostComment Entities
+                List<Tag> updateTagsRequest = this.postMapperDelegate.tagRequestListToTagList(postRequest.tags());
+
+                List<Tag> existingTags =
+                        post.getTags().stream().map(PostTag::getTag).toList();
+
+                // Remove the existing database rows that are no
+                // longer found in the incoming collection (updateTagsRequest)
+                List<Tag> tagsToRemoveList = existingTags.stream()
+                        .filter(tag -> !updateTagsRequest.contains(tag))
+                        .toList();
+                tagsToRemoveList.forEach(post::removeTag);
+
+                List<String> tagNames =
+                        existingTags.stream().map(Tag::getTagName).toList();
+                // Update the existing database rows which can be found
+                // in the incoming collection (updateTagsRequest)
+                List<Tag> newTagsList = updateTagsRequest.stream()
+                        .filter(tag -> !tagNames.contains(tag.getTagName()))
+                        .toList();
+
+                updateTagsRequest.stream()
```

Comment:
@coderabbitai, can you please revisit the logic of updating the tag and suggest better alternative.

Reply:
@coderabbitai, as code is running under single transaction from service method below code will give error

// Clear existing tags and add all requested tags
    post.getTags().clear();
    tagRequests.forEach(req -> post.addTag(tagMap.get(req.name())));

Reply:
@coderabbitai, Using the helper method and 1-n bidirectional mapping from posts. Please rewrite logic to create, remove and modify values from the request with values in db. Handle merging/modify/updating carefully. Rest is working as expected.

Reply:
@coderabbitai, in above code suggestion, there is no need to call db to Find existing tags in DB in a single query, as they are already present in Post Object which is a db object. Please use this and rewrite to avoid extra db calls